packages/core/src/Site/template.ts

```diff
@@ -3,16 +3,21 @@ import path from 'path';
 import * as fsUtil from '../utils/fsUtil';
 
 const requiredFiles = ['index.md', 'site.json', '_markbind/'];
+const omitOnConvertFiles = ['.gitignore'];
 
 const PATH_TO_TEMPLATE = '../../template';
 
 export class Template {
   root: string;
   template: string;
+  toConvert: boolean;
+  omitFiles: Array<string>;
```

Comment:
Are these fields needed since we pass an `overwrite: false` option passed to the `copySyncWithOption` function, which may automatically avoid overwriting an existing .gitignore? (And we only want to omit copying a .gitignore if it already exists)

Reply:
Good point!

This implementation ensures that regardless of pre-existing files, if the `--convert` flag is used, we will entirely skip `.gitignore` (and potentially other files declared in `omitOnConvertFiles` if we want to extend to `package.json` as well later on).

If we instead rely on `overwrite: false` alone, this would mean that we will generate a `.gitignore` when converting documents without a `.gitignore` in their root repository. 

I don't think we came to a conclusion for this as mentioned by @tlylt [here](https://github.com/MarkBind/markbind/issues/1782#issuecomment-1488331079), but you're right that this is a more complex implementation. I can revert these for now, and if we later decide to implement skipping files for `--convert` altogether, we can refer back to these changes 👍 

Does that sound good @jovyntls?

Reply:
Ah I see, sounds good to me - I think it's good to have add gitignore in convert (if there isn't an existing one)

Reply:
Changes reverted in [1297fbe](https://github.com/MarkBind/markbind/pull/2248/commits/1297fbe23477cd06027f0446a03a5d63c91f0548)!